services/src/main/java/org/keycloak/protocol/AuthorizationEndpointBase.java

```diff
@@ -188,16 +189,19 @@ protected AuthenticationSessionModel createAuthenticationSession(ClientModel cli
             UserSessionModel userSession = userSessionCrossDCManager.getUserSessionIfExistsRemotely(manager, realm);
 
             if (userSession != null) {
-                String userSessionId = userSession.getId();
-                rootAuthSession = session.authenticationSessions().createRootAuthenticationSession(userSessionId, realm);
-                authSession = rootAuthSession.createAuthenticationSession(client);
-                logger.debugf("Sent request to authz endpoint. We don't have root authentication session with ID '%s' but we have userSession." +
-                        "Re-created root authentication session with same ID. Client is: %s . New authentication session tab ID: %s", userSessionId, client.getClientId(), authSession.getTabId());
+                if (AuthorizationEndpoint.Action.REGISTER.equals(action)) {
+                    authSession = createNewAuthenticationSession(manager, client);
+
+                    session.sessions().removeUserSession(realm, userSession);
```

Comment:
Few points to this:

1) It may be slightly better to call "AuthenticationManager.backchannelLogout" That one also cleanup cookies, logout all clients from the expired userSession etc.

2) I can think of the scenario like this:
- User is logged
- The registration URL is opened (for example after press "Register" link from the application)
- User then decides to go "Back" from the registration form. For example he pressed "Register" link by mistake in the previous step. But he won't be able to go back as session is already logged-out.

So my vote is instead to handle this in AuthenticationManager.attachSession (which is triggered after registration flow is successfully finished) rather than at the beginning of the flow.
- Now the session is already